src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -0,0 +1,371 @@
+   * ```
+   */
+  viewableEntryTypes: EntryType[] = [];
+
+  private localizationService = new LfLocalizationService();
+
+  constructor(
+    private repoClient: IRepositoryApiClientEx,
+  ) {
+    if (!this.repoClient) {
+      console.error('getParentNodeAsync() _selectedRepositoryClient was undefined');
+      throw new Error(this.localizationService.getString('ERRORS.ENTRY_NOT_FOUND'));
+    }
+  }
+
+  /**
+   * Gets the parent LfTreeNode of the passed in node
+   * @param treeNode: LfTreeNode object to get the parent of
+   * returns - Promise with the parent of the parameter LfTreeNode or undefined if its the root LfTreeNode and does not have a parent
+   * @example
+   * ```ts
+   * // suppose the repository is structured as below
+   * - root
+   *      - DocInRoot
+   *      - FolderInRoot
+   *            - DocInFolderInRoot
+   *            - FolderInFolderInRoot
+   *            - FolderInFolderInRoot1
+   *      - FolderInRoot1
+   *      - FolderInRoot2
+   *      - FolderInRoot3
+   *
+   * const service = new LfRepoTreeNodeService(repoClient);
+   * service.getParentTreeNodeAsync(DocInFolderInRoot) -> FolderInRoot;
+   * ```
+   */
+  async getParentTreeNodeAsync(treeNode: LfRepoTreeNode): Promise<LfRepoTreeNode | undefined> {
+    const repoName: string = await this.repoClient.getCurrentRepoName();
+    const repoId: string = await this.repoClient.getCurrentRepoId();
+    if (treeNode.id === '1') {
+      return undefined;
+    }
+    if (treeNode.path?.endsWith('\\')) {
+      treeNode.path = treeNode.path.slice(0,-1);
+    }
+
+    try {
+      const parentPath = this.getParentPath(treeNode.path);
+      if (parentPath) {
+        const parentEntry: FindEntryResult = await this.repoClient.entriesClient.getEntryByPath({repoId, fullPath: parentPath});
+        if (parentEntry.entry) {
+          return this.createNode(parentEntry.entry, repoName);
+        }
+        else {
+          throw new Error('Parent is not found');
+        }
+      }
+      else {
+        return this.getRootTreeNodeAsync();
+      }
+
+    }
+    catch (err: any) {
+      if (err.errorCode === 9013) {
+        const rootNode = await this.getRootTreeNodeAsync();
+      }
+      else {
+        throw err;
+      }
+    }
+  }
+
+  /**
+   * Requests a page of size 20 containing the children of the parent tree node, sorted by name, ascending.
+   * Only returns the viewable children.
+   * @param folder: LfTreeNode representing the folder to get data from
+   * @param nextPage: string representing the next page requested. If undefined, the first page is returned
+   * @example
+   * ```ts
+   * // suppose the repository is structured as below
+   * - root
+   *      - DocInRoot
+   *      - FolderInRoot
+   *            - DocInFolderInRoot
+   *            - FolderInFolderInRoot
+   *            - FolderInFolderInRoot1
+   *            - FolderInFolderInRoot2
+   *            ...
```

Comment:
I think if we up the page size we don't need quite so many example nodes, this is a very long example

Reply:
looks like page size is still 20 for nextPage
also do we have to set the rest of the parameters when using next page? (select, etc)

Reply:
We don't have to set the rest of the parameters, because the parameters are embedded in the request url return as nextPageLink by the api call